docs/source/links/rest.md

```diff
@@ -0,0 +1,394 @@
+    fieldNameDenormalizer: (key: string) => snake_case(key),
+    typePatcher: {
+      Post: ()=> {
+        bodySnippet...
+      }
+    },
+  });
+```
+
+<h2 id="context">Link Context</h2>
+
+`RestLink` has an [interface `LinkChainContext`](https://github.com/apollographql/apollo-link-rest/blob/1824da47d5db77a2259f770d9c9dd60054c4bb1c/src/restLink.ts#L557-L570) which it uses as the structure of things that it will look for in the `context`, as it decides how to fulfill a specific `RestLink` request. (Please see the [`apollo-link-context`](./context.html) page for a discussion of why you might want this).
+
+* `credentials?: RequestCredentials`: overrides the `RestLink`-level setting for `credentials`. [Values documented here](https://developer.mozilla.org/en-US/docs/Web/API/Request/headers)
+* `headers?: Headers`: Additional headers provided in this `context-link` [Values documented here](https://developer.mozilla.org/en-US/docs/Web/API/Request/headers)
+* `headersToOverride?: string[]` If you provide this array, we will merge the headers you provide in this link, by replacing any matching headers that exist in the root `RestLink` configuration. Alternatively you can use `headersMergePolicy` for more fine-grained customization of the merging behavior.
+* `headersMergePolicy?: RestLink.HeadersMergePolicy`: This is a function that decide how the headers returned in this `contextLink` are merged with headers defined at the `RestLink`-level. If you don't provide this, the headers will be simply appended. To use this option, you can provide your own function that decides how to process the headers. [Code references](https://github.com/apollographql/apollo-link-rest/blob/8e57cabb5344209d9cfa391c1614fe8880efa5d9/src/restLink.ts#L462-L510)
+
+<h3 id="context.headers">Example</h3>
+`RestLink` uses the `headers` field on the [`apollo-link-context`](./context.html) so you can compose other links that provide additional & dynamic headers to a given query. These headers will be merged with
+
+Here is one way to add `headers` to the context:
+
+```js
+const authRestLink = setContext(async () => {
+  const token = await localStorage.getItem("token");
+  return {
+    headers: {
+      Accept: "application/json",
+      Authorization: `Bearer ${token}`
+    }
+  };
+);
+
+const restLink = new RestLink({ uri: "uri" });
+
+const client = new ApolloClient({
+  link: ApolloLink.from([authRestLink, restLink ]),
+  cache: new InMemoryCache(),
+});
+```
+
+<h2 id="order">Link order</h2>
+
+If you are using multiple link types, `restLink` should go before `httpLink`, as `httpLink` will swallow any calls that should be routed through `apollo-link-rest`!
+
+For example:
+
+```js
+const httpLink = createHttpLink({ uri: "server.com/graphql" });
+const restLink = new RestLink({ uri: "api.server.com" });
+
+const client = new ApolloClient({
+  link: ApolloLink.from([authLink, restLink, httpLink, retryLink, errorLink]),
```

Comment:
The ordering should be `[authLink, restLink, retryLink, errorLink, httpLink]`, since http link is terminating and the retry/error links would never run.